src/util.rs

```diff
@@ -54,42 +53,44 @@ pub fn process_arguments(arguments: &[Value]) -> (Argc, *const Value) {
     (arguments.len() as Argc, arguments.as_ptr())
 }
 
-pub fn option_to_slice<'a, T>(option: &'a Option<T>) -> &'a [T] {
+pub fn option_to_slice<T>(option: &Option<T>) -> &[T] {
     match option {
-        &Some(ref v) => unsafe { slice::from_raw_parts(v, 1) },
-        &None => &[],
+        Some(v) => unsafe { slice::from_raw_parts(v, 1) },
+        None => &[],
     }
 }
 
-// Converts a pointer to array of `AnyObject`s to `Vec<AnyObject>`.
-//
-// This function is a helper for callbacks, do not use it directly.
-//
-// It will be moved to other struct, because it is not related to VM itself.
-//
-// # Examples
-//
-// ```no_run
-// use rutie::types::Argc;
-// use rutie::{AnyObject, Boolean, Class, Object, RString, util};
-//
-// #[no_mangle]
-// pub extern fn string_eq(argc: Argc, argv: *const AnyObject, rtself: RString) -> Boolean {
-//     let argv = util::parse_arguments(argc, argv);
-//     let other_string = argv[0].try_convert_to::<RString>().unwrap();
-//
-//     Boolean::new(rtself.to_str() == other_string.to_str())
-// }
-//
-// fn main() {
-//     Class::from_existing("String").define_method("==", string_eq);
-// }
-// ```
-pub fn parse_arguments(argc: Argc, arguments: *const AnyObject) -> Vec<AnyObject> {
+/// Converts a pointer to array of `AnyObject`s to `Vec<AnyObject>`.
+///
+/// This function is a helper for callbacks, do not use it directly.
+///
+/// It will be moved to other struct, because it is not related to VM itself.
+///
+/// # Examples
+///
+/// ```
+/// use rutie::types::Argc;
+/// use rutie::{AnyObject, Boolean, Class, Object, RString, util, VM};
+///
+/// #[no_mangle]
+/// pub extern fn string_eq(argc: Argc, argv: *const AnyObject, rtself: RString) -> Boolean {
+///     let argv = unsafe { util::parse_arguments(argc, argv) };
+///     let other_string = argv[0].try_convert_to::<RString>().unwrap();
+///
+///     Boolean::new(rtself.to_str() == other_string.to_str())
+/// }
+///
+/// fn main() {
+///     # VM::init();
+///     Class::from_existing("String").define_method("==", string_eq);
+/// }
+/// ```
+/// TODO: Made this public function unsafe due to clippy linting, should we just skip the lint?
```

Comment:
Just mark the breaking change in the README.  This note can then be removed.

Reply:
Addressed in 0a743fb3b3d9eedbd431e17bfcdbf4e676c7c74f